packages/utils/crates/utils-types/src/primitives.rs

```diff
@@ -1,211 +1,460 @@
+use std::cmp::Ordering;
 use ethnum::{u256, AsU256};
 use std::ops::{Add, Sub};
-
-pub trait TypeBase: PartialEq + ToString {
-
-    fn to_hex(&self) -> String;
+use std::string::ToString;
+use crate::errors::{Result, GeneralError::ParseError};
+use crate::errors::GeneralError::MathError;
+
+/// Represents an Ethereum address
+#[derive(Clone, Eq, PartialEq, Debug)]
+#[cfg_attr(feature = "wasm", wasm_bindgen::prelude::wasm_bindgen)]
+pub struct Address {
+    addr: [u8; Self::SIZE],
 }
 
-pub trait BaseBalance: TypeBase {
-    const SYMBOL: &'static str;
-
-    fn value(&self) -> &u256;
+#[cfg_attr(feature = "wasm", wasm_bindgen::prelude::wasm_bindgen)]
+impl Address {
+    #[cfg_attr(feature = "wasm", wasm_bindgen::prelude::wasm_bindgen(constructor))]
+    pub fn new(bytes: &[u8]) -> Self {
+        assert_eq!(bytes.len(), Self::SIZE, "invalid length");
+        let mut ret = Address {
+            addr: [0u8; Self::SIZE]
+        };
+        ret.addr.copy_from_slice(bytes);
+        ret
+    }
 
-    fn symbol(&self) -> &str {
-        Self::SYMBOL
+    pub fn to_hex(&self) -> String {
+        hex::encode(self.addr)
     }
 
-    fn serialize_value(&self) -> Box<[u8]> {
-        Box::new(self.value().to_be_bytes())
+    pub fn serialize(&self) -> Box<[u8]> {
+        self.addr.into()
```

Comment:
Should there be a trait for the `Serialize` and `Deserialize` methods? There are quite a few of them, so a trait would make sense.

Reply:
Applies the same as I commented above, trait impls cannot have `#[wasm_bindgen]` annotations on them, which would lead to code duplication for the moment.